### PR TITLE
[6.3] FIX UI test_role locked

### PR DESCRIPTION
**Pull request by ldjebran**
The locked roles have no link to click to view their resources and permissions, the only possible solution is to push the filters button to be redirected to Filters role page to get data from there.
test_positive_clone_builtin was randomly passing when selecting a non locked role, but when randomly select a locked one it fail, the test was refactored to choose randomly one locked and one unlocked role.
 
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test -v tests/foreman/ui/test_role.py::RoleTestCase
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 11 items 
2017-09-11 11:44:43 - conftest - DEBUG - Found WONTFIX in decorated tests ['1156555', '1269196', '1378009', '1245334', '1217635', '1226425', '1147100', '1311113', '1199150', '1204686', '1221971', '1103157', '1230902', '1230865', '1214312', '1079482']

2017-09-11 11:44:43 - conftest - DEBUG - Collected 11 test cases


tests/foreman/ui/test_role.py::RoleTestCase::test_negative_create_with_invalid_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_assign_cloned_role PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_clone_builtin PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_clone_custom PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_filter_admin_user_with_locs PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_filter_admin_user_with_orgs PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_with_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_delete PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_delete_cloned_builtin PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_update_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_update_permission PASSED

================================================== 0 tests deselected ==================================================
============================================= 11 passed in 1016.33 seconds =============================================
```

### CODE REVIEW

File: robottelo/ui/locators/base.py [hunk 1]
```diff
@@ -1279,6 +1279,24 @@
         "not(contains(@class, 'disabled'))]/a"
     ),
     "roles.filters.search": (By.XPATH, "//input[@type='search']"),
+    "roles.filters_button": (
+        By.XPATH,
+        "//td/em[normalize-space(.)='%s']"
+        "/following::td//a[@data-id='aid_filters']"
+        " | //td/span/a[normalize-space(.)='%s']"
+        "/following::td//a[@data-id='aid_filters']"),
```
Comment (abalakh) [inline robottelo/ui/locators/base.py]: `//td[contains(., '%s')]/following-sibling::td//a[@data-id='aid_filters']` works with both styles and doesn't require workarounds like `locators['roles.filters_button'] % (role_name, role_name)` later.

File: robottelo/ui/role.py [hunk 2]
```diff
@@ -83,22 +83,48 @@ def get_resources(self, role_name):
...
+        resources = [
+            resource.text for resource in
+            self.find_elements(locators['role_filters.resources'])
+            ]
+        for res_type in resources:
+            if res_type in resource_types:
+                perm_element = self.find_element(
+                    locators['role_filters.permissions'] % res_type)
+                if perm_element:
+                    permissions[res_type] = perm_element.text.split(', ')
```
Comment (abalakh) [inline robottelo/ui/role.py]: Could you please describe what's going on here? Previously the table contained 1 resource type per line, and for each resource type all the permissions were shown in one cell, separated by comma. That's why we had lines like this (`permissions[res_type] = perm_element.text.split(', ')`).
Now, as far as i can see, we have separate line per each permission, making this code invalid. The fix should have been something like grabbing all the cells where preceding cell (resource type) is equal to `res_type`. 
For me locally the fix is only 3 lines long, but here i can see one `get_permissions` rewritten and extra private method created. What's the reason behind this?

Comment (abalakh) [inline robottelo/ui/role.py]: You should append that list of permissions, not rewrite it. It's possible to create multiple filters with the same resource type, but different permissions. In the end, role will receive all the permissions. E.g.:
Filter 1) Resource type: Organization, permissions: view_organizations, create_organizations.
Filter 2) Resource type: Organization, permissions: edit_organizations, destroy_organizations.
Filter 3) Resource type: Organization, permissions: assign_organizations.

Assigning such role will give user sum of all these permissions (thus, user will receive all possible organization permissions).

**Comment by ldjebran** [comment]
has to fix the branch

**Comment by ldjebran** [comment]
@abalakh changes done
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test -v tests/foreman/ui/test_role.py::RoleTestCase
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2, cov-2.5.1
collected 11 items 
2017-09-14 17:12:10 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_role.py::RoleTestCase::test_negative_create_with_invalid_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_assign_cloned_role PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_clone_builtin PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_clone_custom PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_filter_admin_user_with_locs PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_filter_admin_user_with_orgs PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_create_with_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_delete PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_delete_cloned_builtin PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_update_name PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_update_permission PASSED

============================================= 11 passed in 1119.88 seconds =============================================
```

**Pull request merged**